Gems/Multiplayer/Code/Source/Components/NetworkRigidBodyComponent.cpp

```diff
@@ -75,34 +75,36 @@ namespace Multiplayer
     {
         uint32_t frameId = static_cast<uint32_t>(Multiplayer::GetNetworkTime()->GetHostFrameId());
 
-        AzPhysics::RigidBody* rigidBody = m_physicsRigidBodyComponent->GetRigidBody();
-        rigidBody->SetFrameId(frameId);
-
-        AZ::Transform rewoundTransform;
-        const AZ::Transform& targetTransform = m_transform.Get();
-        const float blendFactor = Multiplayer::GetNetworkTime()->GetHostBlendFactor();
-        if (blendFactor < 1.f)
-        {
-            // If a blend factor was supplied, interpolate the transform appropriately
-            const AZ::Transform& previousTransform = m_transform.GetPrevious();
-            rewoundTransform.SetRotation(previousTransform.GetRotation().Slerp(targetTransform.GetRotation(), blendFactor));
-            rewoundTransform.SetTranslation(previousTransform.GetTranslation().Lerp(targetTransform.GetTranslation(), blendFactor));
-            rewoundTransform.SetUniformScale(AZ::Lerp(previousTransform.GetUniformScale(), targetTransform.GetUniformScale(), blendFactor));
-        }
-        else
+        if (AzPhysics::RigidBody* rigidBody = m_physicsRigidBodyComponent->GetRigidBody())
```

Comment:
Ah, even though this component requires rigidbody I guess it's possible to remove it from the entity after creation, or for OnSyncRewind to get called during entity destruction?
Minor but for large if-wraps sometime I like early return, just so i dont have to remember what if's i'm nested inside of when reading code. Although, it's only a reading problem if it's so long I cant see the top if statement.
Example: 
AzPhysics::RigidBody* rigidBody = m_physicsRigidBodyComponent->GetRigidBody();
if (rigidBody == nullptr)
{
    // Warning maybe?
    return;
}

Reply:
It's the sort of thing that's very possible in a unit test but not that practical otherwise. Still, good to guard against the assumption. That being the case, I don't think a warn is really necessary.